src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -1,70 +1,184 @@
   isXs,
   relay,
   viewer,
 }) => {
+  const tracking = useTracking()
   const { t } = useTranslation()
-  const [value, setValue] = useState("")
+  const [value, setValue] = useState(getSearchTerm(window.location))
+  const [selectedPill, setSelectedPill] = useState<PillType>(TOP_PILL)
+  const [searchIncrement, setSearchIncrement] = useState(0)
 
   const options = extractNodes(viewer.searchConnection)
-  const formattedOptions = options.map(option => {
-    return {
-      text: option.displayLabel!,
-      value: option.displayLabel!,
-      subtitle: getLabel({
-        displayType: option.displayType ?? "",
+  const formattedOptions: SuggionItemOptionProps[] = options.map(
+    (option, index) => {
+      return {
+        text: option.displayLabel!,
+        value: option.displayLabel!,
+        subtitle:
+          getLabel({
+            displayType: option.displayType ?? "",
+            typename: option.__typename,
+          }) ?? "",
+        imageUrl: option.imageUrl!,
+        showArtworksButton: !!option.statuses?.artworks,
+        showAuctionResultsButton: !!option.statuses?.auctionLots,
+        href: option.href!,
         typename: option.__typename,
-      }),
-      imageUrl: option.imageUrl!,
-      showArtworksButton: !!option.statuses?.artworks,
-      showAuctionResultsButton: !!option.statuses?.auctionLots,
-      href: option.href!,
+        item_number: index,
+        item_type: option.displayType!,
+      }
     }
-  })
+  )
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value)
+  useUpdateEffect(() => {
+    tracking.trackEvent({
+      action_type:
+        options.length > 0
+          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+      query: value,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchIncrement])
+
+  const refetch = (value: string, entity?: SearchEntity) => {
+    const entities = entity ? [entity] : []
+    const performanceStart = performance && performance.now()
 
-    //TODO: Add throttle
     relay.refetch(
       {
         hasTerm: true,
-        term: event.target.value,
+        term: value,
+        entities: entities,
       },
       null,
       error => {
         if (error) {
           logger.error(error)
           return
         }
+
+        if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+          reportPerformanceMeasurement(performanceStart)
+        }
+
+        setSearchIncrement(prevCounter => prevCounter + 1)
       }
     )
   }
+  const throttledSearchRequest = useMemo(
+    () => throttle(refetch, SEARCH_THROTTLE_DELAY, { leading: true }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value)
+    throttledSearchRequest(event.target.value)
```

Comment:
**question**: should we also pass the selected pill entity in the second argument?

Reply:
Not sure if we need it here, since we store pill in the state, which we update when switching between pills. That's why when refetching data, pill should be up to date

Reply:
we will always run `refetch` with empty entities when the search query changes (since `entity` is not passed [here](https://github.com/artsy/force/blob/f85fdad385bcfae7c7fdeb016a2f49e59fa42bad/src/Components/Search/NewSearch/NewSearchBarInput.tsx#L143) and an [empty array](https://github.com/artsy/force/blob/f85fdad385bcfae7c7fdeb016a2f49e59fa42bad/src/Components/Search/NewSearch/NewSearchBarInput.tsx#L111) is used by default)

https://user-images.githubusercontent.com/3513494/236439057-96d95e78-0c83-4542-a7dc-339298915026.mp4

Reply:
Weird, I will take a look, thank you